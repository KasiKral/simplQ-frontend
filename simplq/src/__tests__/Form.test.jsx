/* eslint-disable */
import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import Form from '../components/pages/Join/Form';

afterEach(cleanup);

it('renders correctly', () => {
  const { asFragment } = render(<Form />);
  expect(asFragment()).toMatchSnapshot();
});

describe('<Form>', () => {
  it('Name placholder Onchange', () => {
    const { getByPlaceholderText } = render(<Form />);

    const placeholderText = 'Name';
    let userInput = 'Adam';
    fireEvent.change(getByPlaceholderText(placeholderText), { target: { value: userInput } });
    expect(getByPlaceholderText(placeholderText)).toHaveValue(userInput);
  });

  it('Number placholder Onchange', () => {
    const { getByPlaceholderText, getByRole } = render(<Form />);
    const prefix = getByRole('button', { name: /india: \+ 91/i });

    const placeholderText = 'Phone Number';
    let userInput = '8555214486';
    fireEvent.change(getByPlaceholderText(placeholderText), { target: { value: userInput } });
    expect(getByPlaceholderText(placeholderText)).toHaveValue(`+${'855 521 448 6'}`);
    expect(prefix).toBeInTheDocument();
  });
});
