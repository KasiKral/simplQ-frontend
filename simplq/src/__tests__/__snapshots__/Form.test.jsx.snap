// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <div
    class="form"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input"
          placeholder="Name"
          required=""
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          style="padding-left: 8px;"
        >
          <legend
            class="PrivateNotchedOutline-legend-2"
            style="width: 0.01px;"
          >
            <span>
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class=" react-tel-input"
    >
      <input
        class=" form-control"
        name="phone"
        placeholder="Phone Number"
        required=""
        style="width: 100%;"
        type="tel"
        value="+91"
      />
      <div
        class=" flag-dropdown"
      >
        <div
          aria-haspopup="listbox"
          class="selected-flag"
          role="button"
          tabindex="0"
          title="India: + 91"
        >
          <div
            class="flag in"
          >
            <div
              class="arrow"
            />
          </div>
        </div>
      </div>
    </div>
    <button
      class="standard-button"
      type="submit"
    />
  </div>
</DocumentFragment>
`;
